src/Libraries/Microsoft.Extensions.AI/ChatCompletion/LoggingChatClient.cs

```diff
@@ -103,52 +130,63 @@ public override async IAsyncEnumerable<StreamingChatCompletionUpdate> CompleteSt
+                    throw;
+                }
+                catch (Exception ex)
                 {
-                    _logger.LogError(ex, "CompleteStreamingAsync failed.");
+                    LogInvocationFailed(nameof(CompleteStreamingAsync), ex);
                     throw;
                 }
 
                 if (_logger.IsEnabled(LogLevel.Debug))
                 {
                     if (_logger.IsEnabled(LogLevel.Trace))
                     {
-                        _logger.Log(LogLevel.Trace, 0, (update, _jsonSerializerOptions), null, static (state, _) =>
-                            $"CompleteStreamingAsync received update: {JsonSerializer.Serialize(state.update, state._jsonSerializerOptions.GetTypeInfo(typeof(StreamingChatCompletionUpdate)))}");
+                        LogStreamingUpdateSensitive(AsJson(update));
                     }
                     else
                     {
-                        _logger.LogDebug("CompleteStreamingAsync received update.");
+                        LogStreamingUpdate();
                     }
                 }
 
                 yield return update;
             }
 
-            _logger.LogDebug("CompleteStreamingAsync completed.");
+            LogCompleted(nameof(CompleteStreamingAsync));
         }
         finally
         {
             await e.DisposeAsync().ConfigureAwait(false);
         }
     }
 
-    private void LogStart(IList<ChatMessage> chatMessages, ChatOptions? options, [CallerMemberName] string? methodName = null)
-    {
-        if (_logger.IsEnabled(LogLevel.Debug))
-        {
-            if (_logger.IsEnabled(LogLevel.Trace))
-            {
-                _logger.Log(LogLevel.Trace, 0, (methodName, chatMessages, options, this), null, static (state, _) =>
-                    $"{state.methodName} invoked: " +
-                    $"Messages: {JsonSerializer.Serialize(state.chatMessages, state.Item4._jsonSerializerOptions.GetTypeInfo(typeof(IList<ChatMessage>)))}. " +
-                    $"Options: {JsonSerializer.Serialize(state.options, state.Item4._jsonSerializerOptions.GetTypeInfo(typeof(ChatOptions)))}. " +
-                    $"Metadata: {JsonSerializer.Serialize(state.Item4.Metadata, state.Item4._jsonSerializerOptions.GetTypeInfo(typeof(ChatClientMetadata)))}.");
-            }
-            else
-            {
-                _logger.LogDebug($"{methodName} invoked.");
-            }
-        }
-    }
+    private string AsJson<T>(T value) => JsonSerializer.Serialize(value, _jsonSerializerOptions.GetTypeInfo(typeof(T)));
+
+    [LoggerMessage(LogLevel.Debug, "{MethodName} invoked.")]
+    private partial void LogInvoked(string methodName);
+
+    [LoggerMessage(LogLevel.Trace, "{MethodName} invoked: {ChatMessages}. Options: {ChatOptions}. Metadata: {ChatClientMetadata}.")]
+    private partial void LogInvokedSensitive(string methodName, string chatMessages, string chatOptions, string chatClientMetadata);
```

Comment:
As it's already checked if the `LogLevel` is enabled, there's no need for the SGen to generate that check too. So set
```suggestion
    [LoggerMessage(LogLevel.Trace, "{MethodName} invoked: {ChatMessages}. Options: {ChatOptions}. Metadata: {ChatClientMetadata}.", SkipEnabledCheck = true)]
    private partial void LogInvokedSensitive(string methodName, string chatMessages, string chatOptions, string chatClientMetadata);
```
in order to avoid the redundant check be emitted.

Same on other places.

Reply:
I didn't do so because these are only Debug/Trace events, and in the case where such events are enabled, the extra interface call will be dwarfed by all the JSON serialization and other overheads. And while I could just enable it anyway, not all of the call sites are currently guarded, which makes it complicated for someone maintaining the code to know which call sites need guarding and which don't. Which then leads to consistently putting SkipEnabledCheck on all of the log methods and consistently guarding all call sites, but that in turn makes the code more expensive. Seemed best to just keep it simple and pay for the extra enabled check when there's already a ton of overhead / when debugging.